### PR TITLE
Handle execution mode inside of annote and inspect

### PR DESCRIPTION
**Pull request by flavio**
Update the `annotate` and `inspect` commands to handle the new `executionMode` metadata attribute.

This fixes https://github.com/kubewarden/kwctl/issues/57

### CODE REVIEW

File: src/annotate.rs [hunk 1]
```diff
@@ -30,17 +35,76 @@ fn protocol_detector(wasm_path: PathBuf) -> Result<ProtocolVersion> {
...
+
+    Ok(false)
+}
+
 fn prepare_metadata(
     wasm_path: PathBuf,
     metadata_path: PathBuf,
-    detect_protocol_func: impl Fn(PathBuf) -> Result<ProtocolVersion>,
+    kubewarden_protocol_detector_func: impl Fn(PathBuf) -> Result<ProtocolVersion>,
+    rego_detector_func: impl Fn(PathBuf) -> Result<bool>,
```
Comment (ereslibre) [inline src/annotate.rs:63]: I think I prefer to shuffle this data structures a bit, such as:

```rust
enum Backend {
  Unknown,
  OPA,
  OPAGatekeeper,
  KubewardenWapc(ProtocolVersion),
}
```

And we can have a function that performs the heuristics and return a `Result<Backend>`. By decomposing the result, we get the relevant information in each case -- right now, the protocol version in the case of the kubewarden protocol enum value.

The mocking strategy is similar to this one, in that we can pass this heuristics function as an `impl Fn` argument we can mock.

**Review by ereslibre — COMMENTED** [review]
LGTM, some comments regarding some type refactor that I think would be nice.

File: src/annotate.rs (hunk 1 above)
Comment (flavio) [reply src/annotate.rs:63]: I like the proposal, I'll change the code to reflect that. Thanks!

**Comment by flavio** [comment]
@ereslibre I've just pushed a commit which implements the `Backend` enum and the `BackedDetector` struct.

Once you're fine with it, I'll squash the commits around and merge it

**Review by ereslibre — APPROVED** [review]
Thanks a lot, it looks really good :)

**Pull request merged**